### PR TITLE
Node::find

### LINKED ISSUE #226: Optimization for usecases of `try_get_in_node_storage`
Currently code that uses `try_get_in_node_storage` usually has this usage pattern:

```c++
IRI iri = IRI::make("some-iri", temp_node_storage);
...Later
IRI iri_for_use = iri.try_get_in_node_storage(actual_node_storage);
```

This constitutes two relatively expensive node storage accesses.
It could be improved by adding new interfaces to rdf4cpp:
```c++
// returns null if not exists, uses NodeStorage::find_id internally
IRI iri_for_use = IRI::find("some-iri", actual_node_storage); 
```

This would turn the two accesses into a single one.

### PR DESCRIPTION
**Pull request by mcb5637**
close #226

### CODE REVIEW

**Review by liss-h — CHANGES_REQUESTED** [review]
Sorry the issue wasn't formulated very well. This is meant as a function for all node types.

So also Literal::find, BNode::find and so on

File: src/rdf4cpp/rdf/Literal.hpp [hunk 1]
```diff
@@ -334,6 +336,85 @@ class Literal : public Node {
...
+     * searches for a Literal of type T in the specified node storage and returns it.
+     * returns a null Literal, if not found.
+     * if T is inlineable (and not rdf::LangString) always returns the inlined Literal.
+     * @tparam T
+     * @param compatible_value
+     * @param node_storage
+     * @return
+     */
+    template<datatypes::LiteralDatatype T>
+    [[nodiscard]] static Literal find(typename T::cpp_type const &compatible_value,
```
Comment (liss-h) [inline src/rdf4cpp/rdf/Literal.hpp]: Missing templated version that takes `string_view` (== `find_typed`)
+ rename this to `find_typed_from_value`

File: tests/nodes/tests_Node.cpp [hunk 2]
```diff
@@ -200,3 +200,13 @@ TEST_CASE("IRI UUID") {
     CHECK(uuid != uuid2);  // note: non-deterministic but should basically never fail
     CHECK(regex::Regex{"^urn:uuid:[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$"}.regex_match(uuid.identifier()));
 }
+
+TEST_CASE_TEMPLATE("IRI/BlankNode::find", T, IRI, BlankNode) {
```
Comment (liss-h) [inline tests/nodes/tests_Node.cpp]: This test case will fail if we ever implement bnode validation (I think). Please seperate the 2 cases

**Review by liss-h — CHANGES_REQUESTED** [review]
Implementation looks correct to me, just 2 things missing
1. the additional function in literal (added a seperate comment for that)
2. `query::Variable::find`

**Comment by liss-h** [comment]
Also please fix the branch package workflow, it seems that it cannot handle branche names with uppercase letters

File: tests/nodes/tests_Node.cpp (hunk 2 above)
Comment (mcb5637) [reply tests/nodes/tests_Node.cpp]: I assume separating the data of the 2 is enough.
If not, we might want to wait until we know what bnode validation looks like.

**Comment by mcb5637** [comment]
please only merge, when https://github.com/dice-group/cpp-conan-release-reusable-workflow/pull/17 is merged into main and the CI points to main again

**Review by liss-h — APPROVED** [review]
TODO:
- [x] point workflow to main when merged

**Pull request merged**